### PR TITLE
Autopsy Addition: Now detective can actually solve crime![READY]

### PR DESCRIPTION
**Pull request by Djiq**
## About The Pull Request

This PR adds a surgery we all needed.. Autopsy! It allows you to uncover how someone died.

## Why It's Good For The Game

It will makes solving crime more engaging.

## Changelog
:cl:
add: Added new surgery: Autopsy!
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
does detective have access to the surgery room in brig

**Comment by EdgeLordExe** [comment]
> does detective have access to the surgery room in brig

i have no idea and frankly im not going to bother with it

File: code/modules/antagonists/changeling/powers/mutations.dm [hunk 1]
```diff
@@ -153,6 +153,7 @@
 	item_state = "arm_blade"
 	lefthand_file = 'icons/mob/inhands/antag/changeling_lefthand.dmi'
 	righthand_file = 'icons/mob/inhands/antag/changeling_righthand.dmi'
+	autopsy_description = "grivious lacerations"
```
Comment (spessbandit) [inline code/modules/antagonists/changeling/powers/mutations.dm]: grievous

File: code/modules/hydroponics/hydroitemdefines.dm [hunk 2]
```diff
@@ -91,6 +91,7 @@
 	item_state = "hatchet"
 	lefthand_file = 'icons/mob/inhands/equipment/hydroponics_lefthand.dmi'
 	righthand_file = 'icons/mob/inhands/equipment/hydroponics_righthand.dmi'
+	autopsy_description = "chopped wounds"
```
Comment (spessbandit) [inline code/modules/hydroponics/hydroitemdefines.dm]: chop

File: code/game/objects/items/twohanded.dm [hunk 3]
```diff
@@ -219,6 +219,7 @@
 	icon_state = "fireaxe0"
 	lefthand_file = 'icons/mob/inhands/weapons/axes_lefthand.dmi'
 	righthand_file = 'icons/mob/inhands/weapons/axes_righthand.dmi'
+	autopsy_description = "chopped wounds"
```
Comment (spessbandit) [inline code/game/objects/items/twohanded.dm]: chop

File: code/modules/surgery/autopsy.dm [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+		if(istype(O,/obj/item))
+			var/obj/item/I = O
+			to_chat(user,"<span class='notice'>Analysis shows that [target] was attacked by a  [I.sharpness == IS_BLUNT ? "blunt" : "sharp"] object causing [I.autopsy_description != null ? "[I.autopsy_description]" : " [I.force > 10 ? "deep wounds" : "shallow wounds" ] "]  for [I.force] of [I.damtype] damage </span>")
+		if(istype(O,/obj/projectile))
+			var/obj/projectile/P = O
+			to_chat(user,"<span class='notice'>Analysis shows that [target] was hit by a [P.flag == "bomb" ? "explosion-like projectile" : "[P.flag]" ] dealing [P.damage] [P.damage_type]  </span>")
+	return ..()
+
+/datum/surgery_step/autopsy/failure(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	user.visible_message("<span class='notice'>[user]'s finger slips destroying the data!</span>", "<span class='notice'>Your finger slips while researching destroying the data!</span>")
```
Comment (ATH1909) [inline code/modules/surgery/autopsy.dm]: need commas

**Comment by ATH1909** [comment]
Ah yes, the iconic detective surgery that detectives can't perform without breaking into Medbay or into a restricted area of Sec. Brilliant.

I kind of feel like you should reduce the number of steps in this surgery, and maybe make the detective's scanner a usable tool for the "autopsy" step?

File: code/modules/surgery/autopsy.dm [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+	name = "analyze cause of death"
+	implements = list(TOOL_HEMOSTAT = 90, TOOL_WIRECUTTER = 35, /obj/item/stack/packageWrap = 15, /obj/item/stack/cable_coil = 5)
+	time = 90
+	experience_given = 20
+
+/datum/surgery_step/autopsy/preop(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	user.visible_message("<span class='notice'>[user] starts analyzing [target].</span>", "<span class='notice'>You start analyzing  [target].</span>")
+
+/datum/surgery_step/autopsy/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery, default_display_results = FALSE)
+	for(var/obj/O in target.damage_record)
```
Comment (ATH1909) [inline code/modules/surgery/autopsy.dm]: add a do_after delay or something to these, or else the surgeon's chat could get spammed with messages if someone died a death of a thousand cuts

File: code/_onclick/item_attack.dm [hunk 6]
```diff
@@ -70,7 +70,8 @@
 
 	user.do_attack_animation(M)
 	M.attacked_by(src, user)
-
+	if(force)
+		M.damage_record += src
```
Comment (ATH1909) [inline code/_onclick/item_attack.dm]: what about (handless) simplemobs and punches? shouldn't they also get logged?

**Comment by cacogen** [comment]
The coroner does the autopsy and the detective gets a copy of the autopsy report. In this case, it'd be a doctor. Or just ask the AI to let you in. Or break in.

**Comment by HomocidlePancakes** [comment]
No, fuck the surgery. Either add a new device or add this in to forensic scanner.

**Comment by cacogen** [comment]
You could make the surgery computer update the patient's medical records so the detective can access them.

**Comment by Tlaltecuhtli** [comment]
make the autopsy step use forsenic for 100% and standard tools for 20%

File: code/_onclick/item_attack.dm (hunk 6 above)
Comment (EdgeLordExe) [reply code/_onclick/item_attack.dm]: oh yeah right i have forgotten bout those

**Comment by EdgeLordExe** [comment]
> Ah yes, the iconic detective surgery that detectives can't perform without breaking into Medbay or into a restricted area of Sec. Brilliant.
> 
> I kind of feel like you should reduce the number of steps in this surgery, and maybe make the detective's scanner a usable tool for the "autopsy" step?

detectives already ask for 50% of their stuff from the warden, making them ask for the surgery bag won't really affect anything. And yeah i agree I will make it require the scanner

**Comment by XDTM** [comment]
Should probably add a description for wounds from a welding tool, since it's fairly common.

Does healing clear the list? It would be odd to heal up and still have all the evidence of past trauma.

File: code/modules/surgery/autopsy.dm [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+		if(istype(O,/obj/item))
+			var/obj/item/I = O
+			to_chat(user,"<span class='notice'>Analysis shows that [target] was attacked by a  [I.sharpness == IS_BLUNT ? "blunt" : "sharp"] object causing [I.autopsy_description != null ? "[I.autopsy_description]" : " [I.force > 10 ? "deep wounds" : "shallow wounds" ] "]  for [I.force] of [I.damtype] damage </span>")
+		if(istype(O,/obj/projectile))
+			var/obj/projectile/P = O
+			to_chat(user,"<span class='notice'>Analysis shows that [target] was hit by a [P.flag == "bomb" ? "explosion-like projectile" : "[P.flag]" ] dealing [P.damage] [P.damage_type]  </span>")
```
Comment (XDTM) [inline code/modules/surgery/autopsy.dm]: Specific damage numbers are a bit immersion-breaking, as well as allowing people to metagame the murder weapon if it's something with a specific damage number (24 means fireaxe, for example).
I'd suggest only having thresholds for shallow wounds, deep wounds, maybe devastating wounds for huge bursts of damage.

Comment (EdgeLordExe) [reply code/modules/surgery/autopsy.dm]: that is a good idea!

**Comment by EdgeLordExe** [comment]
> Should probably add a description for wounds from a welding tool, since it's fairly common.
> 
> Does healing clear the list? It would be odd to heal up and still have all the evidence of past trauma.

only reviving clears the list.

**Comment by XDTM** [comment]
> 
> 
> > Should probably add a description for wounds from a welding tool, since it's fairly common.
> > Does healing clear the list? It would be odd to heal up and still have all the evidence of past trauma.
> 
> only reviving clears the list.

Could end up in situations where someone gets into a fight, gets healed, then half an hour later dies of poison but still appears beaten up when doing a biopsy.

**Comment by HomocidlePancakes** [comment]
>detectives already ask for 50% of their stuff from the warden, making them ask for the surgery bag won't really affect anything

“X is bad, but Y is bad so it justifies X”. Come on, dude. You are half-arsing this.

File: code/modules/surgery/autopsy.dm (hunk 7 above)
Comment (EdgeLordExe) [reply code/modules/surgery/autopsy.dm]: implemented

**Comment by EdgeLordExe** [comment]
> > detectives already ask for 50% of their stuff from the warden, making them ask for the surgery bag won't really affect anything
> 
> “X is bad, but Y is bad too, so it justifies X”. Come on, dude. You are half-arsing this.

look i really really really don't want to deal with mapping

**Comment by EdgeLordExe** [comment]
> > > Should probably add a description for wounds from a welding tool, since it's fairly common.
> > > Does healing clear the list? It would be odd to heal up and still have all the evidence of past trauma.
> > 
> > 
> > only reviving clears the list.
> 
> Could end up in situations where someone gets into a fight, gets healed, then half an hour later dies of poison but still appears beaten up when doing a biopsy.

i guess, but if you think about it even if you get into brawl you still have some signs that this happened a few days later, and rounds take like 3 in game days?

File: code/_onclick/item_attack.dm (hunk 6 above)
Comment (EdgeLordExe) [reply code/_onclick/item_attack.dm]: implemented

File: code/modules/surgery/autopsy.dm (hunk 5 above)
Comment (EdgeLordExe) [reply code/modules/surgery/autopsy.dm]: reworked how it shows up -> now it will be in a note. resolved

File: code/modules/surgery/autopsy.dm (hunk 4 above)
Comment (EdgeLordExe) [reply code/modules/surgery/autopsy.dm]: done

File: code/modules/surgery/autopsy.dm (hunk 5 above)
Comment (EdgeLordExe) [reply code/modules/surgery/autopsy.dm]: reworked how it shows up -> now it will be in a note. resolved

File: code/modules/antagonists/changeling/powers/mutations.dm (hunk 1 above)
Comment (EdgeLordExe) [reply code/modules/antagonists/changeling/powers/mutations.dm]: done

File: code/modules/surgery/autopsy.dm [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+	N.name = text("AR-[] 'Autopsy Record'",frNum)
+	N.info = text("<center><B>Autopsy Record - (FR-[])</B></center><HR><BR>",frNum)
+	N.info += "<HR><B>Notes:</B><BR>"
+
+	for(var/O in target.damage_record)
+
+		if(istype(O,/obj/item))
+			var/obj/item/I = O
+			N.info += "Analysis shows that [target] was attacked by a  [I.sharpness == IS_BLUNT ? "blunt" : "sharp"] object causing [I.autopsy_description != null ? "[I.autopsy_description] of " : " [I.force > 10 ? "[I.force > 20 ? "devastating" : "deep" ]" : "shallow" ] "] [I.damtype] damage.<BR><BR>" //double breaks for easier reading
```
Comment (XDTM) [inline code/modules/surgery/autopsy.dm]: If there is an autopsy description the message will be something like "Lacerations of brute damage", which sounds a bit awkward. I'd change it to either just "Lacerations", or "Lacerations causing [severity] [type] damage"

File: code/modules/hydroponics/hydroitemdefines.dm (hunk 2 above)
Comment (EdgeLordExe) [reply code/modules/hydroponics/hydroitemdefines.dm]: done

File: code/game/objects/items/twohanded.dm (hunk 3 above)
Comment (EdgeLordExe) [reply code/game/objects/items/twohanded.dm]: done

File: code/modules/surgery/autopsy.dm (hunk 8 above)
Comment (EdgeLordExe) [reply code/modules/surgery/autopsy.dm]: hmm that is a good proposition

**Comment by XDTM** [comment]
After this PR we can also build on this surgery to add all sorts of damage sources, like poison, diseases, or explosives, for deep coroner gameplay

File: code/modules/surgery/autopsy.dm (hunk 8 above)
Comment (EdgeLordExe) [reply code/modules/surgery/autopsy.dm]: implemented

**Comment by HomocidlePancakes** [comment]
>look i really really really don't want to deal with mapping

Move the functionality to forensic scanner then. This is a good feature, but you are handicapping it.

**Comment by EdgeLordExe** [comment]
> > look i really really really don't want to deal with mapping
> 
> Move the functionality to forensic scanner then. This is a good feature, but you are handicapping it.

well if i moved autopsy to forensic scanner it wouldnt be autopsy.. would it?

**Comment by Qustinnus** [comment]
daily reminder that what we really need is wound datums.

**Comment by Krysonism** [comment]
Make the surgery like scalpel, retractor, detective scanner.

 And give det retractor + scalpel in his room.

On paradise the coroner autopsies all the corpses and calls the det if he finds anything or the det asks when he stops by.

Alternatively, add a roller bed or surgical table to the morgue and a couple of tools and let the det come and do it on site. This might be the best option.

**Comment by XDTM** [comment]
In fiction it usually isn't the detective doing the autopsy, though

**Comment by EdgeLordExe** [comment]
> In fiction it usually isn't the detective doing the autopsy, though

yeah i know but we dont really have a role closer to a forensic specialist or a coroner

**Comment by XDTM** [comment]
Doctor sounds close enough

**Comment by EdgeLordExe** [comment]
> Make the surgery like scalpel, retractor, detective scanner.
> 
> And give det retractor + scalpel in his room.
> 
> On paradise the coroner autopsies all the corpses and calls the det if he finds anything or the det asks when he stops by.
> 
> Alternatively, add a roller bed or surgical table to the morgue and a couple of tools and let the det come and do it on site. This might be the best option.

i will implement this change, ill add retractor and scalpel to his wardrobe

**Comment by EdgeLordExe** [comment]
> Doctor sounds close enough

well doctors can do that surgery , they just have a lower chance of succeding

**Comment by HomocidlePancakes** [comment]
fug. I guess just giving him a scalpel and a retractor works.

**Comment by ma44** [comment]
It seems like damage_record is only actually cleared whenever a full revive happens, what about for the people that get punched to critical but healed up, still showing off their punch wounds should they die a later death?

**Comment by AWalton** [comment]
Brig Physician being ported from Bee would fit in well with this, and is just generally a good new role to add.

**Comment by Krysonism** [comment]
> Brig Physician being ported from Bee would fit in well with this, and is just generally a good new role to add.

I don't like brig Phys. The role only serves to give all the officers x-ray eyes. Submitting yourself to medical treatment should be able to back fire on you.

Coroner is a comfy role but I don't think there is enough for a coroner to do on/TG/ especially since maintainers are set on making organ harvesting redundant.

I say let the detective do it until there is the need for a coroner.

**Comment by EdgeLordExe** [comment]
> It seems like damage_record is only actually cleared whenever a full revive happens, what about for the people that get punched to critical but healed up, still showing off their punch wounds should they die a later death?

this is something i have to figure out, but there is no clean way of doing it without wound datums (which im not going to add since adding em would be a lot of work and this a small scale pr), so for now damage_record will be cleared only when revived

**Comment by ExcessiveUseOfCobblestone** [comment]
Idea is pretty rad.

Is there not a way to have this work with attack logs? I'm not very keen on adding pretty much a duplicate of said logs for the sake of an optional surgery. This certainly doesn't need to be on the mob.

I'm also slightly concerned with specific entries for each item. I want entries that sweep several items so it lowers the potential items without pinpointing it to a specific one, preferably in an automated way. IE sharp objects should show lacerations, force can determine "deep" or not.

**Comment by EdgeLordExe** [comment]
> Idea is pretty rad.
> 
> Is there not a way to have this work with attack logs? I'm not very keen on adding pretty much a duplicate of said logs for the sake of an optional surgery. This certainly doesn't need to be on the mob.
> 
> I'm also slightly concerned with specific entries for each item. I want entries that sweep several items so it lowers the potential items without pinpointing it to a specific one, preferably in an automated way. IE sharp objects should show lacerations, force can determine "deep" or not.

Regarding logs, i feel your pain but afaik we dont parse combat logs anywhere in the code so i have no idea on how to actually do such a thing, and regarding the second point i wanted make certain very good weapons to leave quite easy to distinguish wounds , so when you kill someone in a cheesy way it is easy to tell what killed them. It is a tradeoff really ease of killing but easier to be found out

**Comment by ExcessiveUseOfCobblestone** [comment]
these items are usually already balanced with that in mind already though (such as unique sounds, being loud, very visible, etc.). This is like 20 minutes after the fact and you found the body while the traitor is ~~continuing their murder spree~~ hiding.

**Comment by EdgeLordExe** [comment]
> these items are usually already balanced with that in mind already though (such as unique sounds, being loud, very visible, etc.). This is like 20 minutes after the fact and you found the body while the traitor is ~continuing their murder spree~ hiding.

well i guess that's true , but the whole point of this surgery is to find out who is the killer in those edge cases where the detective finds a random dead body in maint with no fingerprints

**Comment by ATH1909** [comment]
Don't forget to give the detective access to a cautery too (if the surgery ends with a cautery step).

**Comment by EdgeLordExe** [comment]
> Don't forget to give the detective access to a cautery too (if the surgery ends with a cautery step).

i already did that

**Comment by MrDoomBringer** [comment]
> > Make the surgery like scalpel, retractor, detective scanner.
> > And give det retractor + scalpel in his room.
> > On paradise the coroner autopsies all the corpses and calls the det if he finds anything or the det asks when he stops by.
> > Alternatively, add a roller bed or surgical table to the morgue and a couple of tools and let the det come and do it on site. This might be the best option.
> 
> i will implement this change, ill add retractor and scalpel to his wardrobe



> > Doctor sounds close enough
> 
> well doctors can do that surgery , they just have a lower chance of succeding

Why give the detective surgery tools? This feature seems like a perfect fit for encouraging interdepartmental game play (ie. the detective asking a doctor to perform the surgery, with the surgery requiring the detective's forensic scanner as a step). Don't just give the detective the tools - at least make him go to medbay and ask for them.

Encouraging players to work together breeds paranoia, etc. which this whole gayme is about.

**Comment by EdgeLordExe** [comment]
> > > Make the surgery like scalpel, retractor, detective scanner.
> > > And give det retractor + scalpel in his room.
> > > On paradise the coroner autopsies all the corpses and calls the det if he finds anything or the det asks when he stops by.
> > > Alternatively, add a roller bed or surgical table to the morgue and a couple of tools and let the det come and do it on site. This might be the best option.
> > 
> > 
> > i will implement this change, ill add retractor and scalpel to his wardrobe
> 
> > > Doctor sounds close enough
> > 
> > 
> > well doctors can do that surgery , they just have a lower chance of succeding
> 
> Why give the detective surgery tools? This feature seems like a perfect fit for encouraging interdepartmental game play (ie. the detective asking a doctor to perform the surgery, with the surgery requiring the detective's forensic scanner as a step). Don't just give the detective the tools - at least make him go to medbay and ask for them.
> 
> Encouraging players to work together breeds paranoia, etc. which this whole gayme is about.

 bridge already has the tools but people were screeching bout it , i think ill implement that

**Comment by cacogen** [comment]
Few players do their jobs in a cooperative and timely way which makes
interdepartmental gameplay a case of coder expectations vs. reality.

But giving the Detective surgical instruments is dumb.

You can have a doctor do the surgery. If the doctors aren’t there or don’t
listen you have a huge amount of options. A hardboiled detective like
yourself lets nothing come between you and solving the case. You can wait
for a door to be left open, sneak in and do the surgery yourself. You can
ask the AI to let you in. You can break in. You can get tools from cargo.
You can do it ghetto. You have plenty of options.
-- 
Sent from my phone

**Comment by EdgeLordExe** [comment]
> Few players do their jobs in a cooperative and timely way which makes interdepartmental gameplay a case of coder expectations vs. reality. But giving the Detective surgical instruments is dumb. You can have a doctor do the surgery. If the doctors aren’t there or don’t listen you have a huge amount of options. A hardboiled detective like yourself lets nothing come between you and solving the case. You can wait for a door to be left open, sneak in and do the surgery yourself. You can ask the AI to let you in. You can break in. You can get tools from cargo. You can do it ghetto. You have plenty of options. -- Sent from my phone

that is actually a very good answer lemme delet those tools

**Comment by cacogen** [comment]
did i mention it's a nokia n-gage

**Comment by EdgeLordExe** [comment]
> did i mention it's a nokia n-gage

did anyone ask about that? why would anyone care?

**Comment by Yenwodyah** [comment]
Would be nice if you could tell the age of the wounds too. Like "punched 20 minutes ago, sliced 5 minutes ago, burned 5 minutes ago". Lets you get a sense of which injuries lead to the death, and it's something that real autopsies try to determine as well.

**Comment by cacogen** [comment]
>did anyone ask about that? why would anyone care?

cause its a cool phone and i wanted to make u jealous

-- 
Sent from my N-Gage

**Comment by cacogen** [comment]
\>did anyone ask about that? why would anyone care?

cause its a cool phone and i wanted to make u jealous
\-- 
Sent from my N-Gage

**Review by AnturK — CHANGES_REQUESTED** [review]
Bad, you're adding references that are never cleared and which you don't really need.
On top of that any change post damage to the weapon (including destroy) will show during the autopsy.
Also this kinda duplicates the log code but i suppose that can be improved.

**Comment by EdgeLordExe** [comment]
> Bad, you're adding references that are never cleared and which you don't really need.
> On top of that any change post damage to the weapon (including destroy) will show during the autopsy.
> Also this kinda duplicates the log code but i suppose that can be improved.

Could you provide me with some way of fixing the issue you pointed out? An example of how it would look if it was done in a better way , or something like that?

File: code/modules/surgery/autopsy.dm [hunk 9]
```diff
@@ -26,24 +30,50 @@
...
 	N.info += "<HR><B>Notes:</B><BR>"
 
-	for(var/O in target.damage_record)
+	for(var/datum/autopsy_record/AR in target.damage_record)
 
-		if(istype(O,/obj/item))
-			var/obj/item/I = O
-			N.info += "Analysis shows that [target] was attacked by a  [I.sharpness == IS_BLUNT ? "blunt" : "sharp"] object [I.autopsy_description != null ? "dealing [I.autopsy_description]" : "" ] causing [I.force > 10 ? "[I.force > 20 ? "devastating" : "deep" ]" : "shallow" ]  [I.damtype] damage.<BR><BR>" //double breaks for easier reading
+		if(AR.wound_type == WEAPON_WOUND)
+			N.info += "Analysis shows that [target] was attacked by a  [AR.sharpness == IS_BLUNT ? "blunt" : "sharp"] object [AR.special_description != null ? "dealing [AR.special_description]" : "" ] causing [AR.damage > 10 ? "[AR.damage > 20 ? "devastating" : "deep" ]" : "shallow" ]  [AR.damage_type] damage.<BR><BR>" //double breaks for easier reading
```
Comment (XDTM) [inline code/modules/surgery/autopsy.dm]: considering the word is actually damage and not wounds, i'd rephrase the tiers into minor damage, major damage, and devastating damage

**Comment by Krysonism** [comment]
Having to fetch a doctor each time you want to do an autopsy is not good. Doctors are not supposed to investigate crime, they are busy with actually healing people.

The detective even has his own morgue for gods sake.

**Comment by EdgeLordExe** [comment]
> Having to fetch a doctor each time you want to do an autopsy is not good. Doctors are not supposed to investigate crime, they are busy with actually healing people.
> 
> The detective even has his own morgue for gods sake.

there is already a set of tools in brig

**Comment by AnturK** [comment]
In short store the result text not reference to the mob, and since you want this pretty much all in same places log_combat is used, you can modify that with new parameter that tells it to generate the autopsy log. Ping me on discord if you need some more specific advice.

**Comment by EdgeLordExe** [comment]
> In short store the result text not reference to the mob, and since you want this pretty much all in same places log_combat is used, you can modify that with new parameter that tells it to generate the autopsy log. Ping me on discord if you need some more specific advice.

i decided to create a new datum which houses all the info i need, but i dont know if it is a good way of doing so. Any opinions on it? it is already pushed to here so you can just tell me if i can improve this system , or just scrap it entirely and do something else

**Comment by cacogen** [comment]
>The detective even has his own morgue for gods sake.

yeah i'd forgotten about this you should be able to be the guy
on the other hand,  making the surgery require a scanner is dumb shit

**Comment by EdgeLordExe** [comment]
> > The detective even has his own morgue for gods sake.
> 
> yeah i'd forgotten about this you should be able to be the guy
> on the other hand, making the surgery require a scanner is dumb shit

making it require a scanner for 100% chance of sucess makes it clearly noticable it is primarily for sec/det

**Comment by cacogen** [comment]
>for 100% chance of sucess

i guess

**Comment by Dmp1987** [comment]
> 
> 
> > > Should probably add a description for wounds from a welding tool, since it's fairly common.
> > > Does healing clear the list? It would be odd to heal up and still have all the evidence of past trauma.
> > 
> > 
> > only reviving clears the list.
> 
> Could end up in situations where someone gets into a fight, gets healed, then half an hour later dies of poison but still appears beaten up when doing a biopsy.

Maybe a sort of timestamp to the items in the list with different responses for <5m, <10m, 20, 30 etc
like "these wounds seem fresh" // "the wounds have healed a bit" // "theres fresh / old scarring"

**Comment by Ty-the-Smonk** [comment]
Have a security medical doctor

File: code/_onclick/item_attack.dm [hunk 10]
```diff
@@ -67,10 +67,10 @@
 
 	if(force && M == user && user.client)
 		user.client.give_award(/datum/award/achievement/misc/selfouch, user)
-
+	var/datum/autopsy_record/AR = new /datum/autopsy_record(src,"weapon",damtype,force,name,sharpness,autopsy_description,FALSE)
```
Comment (optimumtact) [inline code/_onclick/item_attack.dm]: does this really need to be a datum?

I'm worried about memory usage if we're adding datums to lists on every single click on a mob
